environments/gravity.py

```diff
@@ -139,28 +145,35 @@ def _draw(self, res=32, color=True):
         q = self._rollout.reshape(2, self.n_objects, 2, -1)[0]
         length = q.shape[-1]
         vid = np.zeros((length, res, res, 3), dtype='float')
+        background_color = self._get_default_background_color()
+        ball_colors = self._get_default_ball_color()
         space_res = 2.*self.get_world_size()/res
+        if self.n_objects == 2:
+            factor = 0.55
+        else:
+            factor = 0.25
         for t in range(length):
             for n in range(self.n_objects):
                 brush = self.colors[n]
                 if brush == 'y':
                     vid[t] = cv2.circle(vid[t],
                                         self._world_to_pixels(
                                             q[n, 0, t], q[n, 1, t], res),
-                                        int(self.mass[n]*.55/space_res), (1., 1., 0.), -1)
+                                        int(self.mass[n]*factor/space_res), ball_colors[0], -1)
                 elif brush == 'r':
                     vid[t] = cv2.circle(vid[t],
                                         self._world_to_pixels(
                                             q[n, 0, t], q[n, 1, t], res),
-                                        int(self.mass[n]*.55/space_res), (1., 0., 0.), -1)
+                                        int(self.mass[n]*factor/space_res), ball_colors[1], -1)
                 elif brush == 'g':
                     vid[t] = cv2.circle(vid[t],
                                         self._world_to_pixels(
                                             q[n, 0, t], q[n, 1, t], res),
-                                        int(self.mass[n]*.55/space_res), (0., 1., 0.), -1)
-            vid[t] = cv2.blur(vid[t], (3, 3))
+                                        int(self.mass[n]*factor/space_res), ball_colors[2], -1)
+            if n_objects == 2:
+                vid[t] = cv2.blur(cv2.blur(vid[t], (2, 2)), (2, 2))
         if color:
-            vid += 80./255.
+            vid[:] += background_color
```

Comment:
If you add the background_color to all values, don't you modify the pre-set ball colors?

Reply:
Yep, the real ball color variable is the addition of the background and the ball color.

Reply:
I will change the addition of the background to be in the initialization of the vid array

Reply:
well, i have been thinking and it's better to add the background afterwards. if you think of it the B channel is always constant. Therefore, when you add an element you need to specify the B channel (which will be the B component of the background). You can set the B component of the elements to 0 if you add the background in the end.

Reply:
Ok, looks good! Thanks for the clarification :)